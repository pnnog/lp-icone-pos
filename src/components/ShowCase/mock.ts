import mockCourses from 'components/CourseCardSlider/mock';
import mockPacientHighlight from 'components/PacientHighlight/mock';
import mockNews from 'components/NewsSlider/mock';

export default {
  courses: mockCourses.slice(0, 1),
  pacientHighlight: mockPacientHighlight,
  news: mockNews.slice(0, 1),
  heading: 'heading teste'
};
