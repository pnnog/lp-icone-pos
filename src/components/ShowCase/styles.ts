import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import Container from 'components/Container';
import * as HeadingStyles from 'components/Heading/styles';
import * as HighlightStyles from 'components/Highlight/styles';
import * as NewsCardSlider from 'components/NewsSlider/styles';

export const Wrapper = styled(Container).attrs({ as: 'section' })`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xlarge};

    > p {
      color: ${theme.colors.black};
      font-size: ${theme.font.sizes.medium};
      font-weight: ${theme.font.regular};
      line-height: 120%;
    }

    ${media.greaterThan('medium')`
      margin-bottom: calc(${theme.spacings.xlarge} * 2 );

      >p{
        font-size: ${theme.font.sizes.xlarge};
        line-height: 120%;
    }

    `}

    ${HeadingStyles.Wrapper} {
      margin-bottom: ${theme.spacings.xsmall};

      ${media.greaterThan('medium')`
        margin-bottom: ${theme.spacings.medium};
    `};
    }

    ${HighlightStyles.Wrapper} {
      ${media.lessThan('medium')`
          margin: 0 calc(-${theme.grid.gutter} / 4);
      `}
    }
  `}
`;
