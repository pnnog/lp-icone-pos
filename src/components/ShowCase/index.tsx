import * as S from './styles';

import Heading from 'components/Heading';
import NewsSlider from 'components/NewsSlider';
import { NewsCardProps } from 'components/NewsCard';
import PacientHighlight, {
  PacientHighlightProps
} from 'components/PacientHighlight';

export type ShowCaseProps = {
  heading?: string;
  news?: NewsCardProps[];
  pacientHighlight?: PacientHighlightProps;
  children?: React.ReactNode;
};
const ShowCase = ({
  heading,
  news,
  pacientHighlight,
  children
}: ShowCaseProps) => (
  <S.Wrapper>
    {!!heading && <Heading>{heading} </Heading>}
    {!!children && <p>{children}</p>}
    {!!news && <NewsSlider items={news} />}
    {!!pacientHighlight && <PacientHighlight {...pacientHighlight} />}
  </S.Wrapper>
);

export default ShowCase;
