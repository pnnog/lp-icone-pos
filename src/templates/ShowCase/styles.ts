import styled, { css } from 'styled-components';
import media from 'styled-media-query';

import Container from 'components/Container';
import * as HeadingStyles from 'components/Heading/styles';
import * as CoursesStyles from 'components/CourseCardSlider/styles';

export const Wrapper = styled(Container).attrs({ as: 'section' })`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.xlarge};

    ${media.greaterThan('medium')`
    margin-bottom: calc(${theme.spacings.xlarge} * 2 );

    `}

    ${HeadingStyles.Wrapper} {
      margin-bottom: ${theme.spacings.xsmall};

      ${media.greaterThan('medium')`
        margin-bottom: ${theme.spacings.medium};
        margin-top: calc(${theme.spacings.large}*2);
    `};
    }

    ${CoursesStyles.Wrapper} {
      ${media.lessThan('medium')`
          margin-right: calc(-${theme.grid.gutter} / 4);
          `}
    }
  `}
`;
