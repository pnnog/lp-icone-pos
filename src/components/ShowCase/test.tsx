import '../../../.jest/match-media-mock';
import { screen } from '@testing-library/react';

import ShowCase, { ShowCaseProps } from '.';
import { renderWithTheme } from 'utils/tests/helper';

import mockNews from 'components/NewsSlider/mock';
import mockPacientHighlight from 'components/PacientHighlight/mock';

const props: ShowCaseProps = {
  heading: 'title test',
  news: mockNews,
  pacientHighlight: mockPacientHighlight
};

jest.mock('components/Heading', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Heading"></div>;
    }
  };
});

jest.mock('components/Card', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Card"></div>;
    }
  };
});

jest.mock('components/NewsSlider', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock News Slider"></div>;
    }
  };
});

jest.mock('components/PacientHighlight', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Pacient Highlight"></div>;
    }
  };
});

describe('<ShowCase />', () => {
  it('should render ShowCase', () => {
    const { container } = renderWithTheme(
      <ShowCase {...props}>children test</ShowCase>
    );
    expect(screen.getByTestId('Mock Heading')).toBeInTheDocument();
    expect(screen.getByTestId('Mock News Slider')).toBeInTheDocument();
    expect(screen.getByTestId('Mock Pacient Highlight')).toBeInTheDocument();
    expect(screen.getByText(/children test/i)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
