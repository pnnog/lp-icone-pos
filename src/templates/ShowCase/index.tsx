import * as S from './styles';

import CourseCardSlider from 'components/CourseCardSlider';
import { CourseCardProps } from 'components/CourseCard';
import ButtonLink, { ButtonLinkProps } from 'components/ButtonLink';
import Heading from 'components/Heading';
import NewsSlider from 'components/NewsSlider';
import { NewsCardProps } from 'components/NewsCard';

export type ShowCaseProps = {
  heading?: string;
  courses?: CourseCardProps[];
  news?: NewsCardProps[];
  button?: boolean;
  description?: string;
};
const ShowCase = ({
  heading,
  courses,
  news,
  description,
  button
}: ShowCaseProps) => (
  <S.Wrapper>
    {!!heading && <Heading>{heading} </Heading>}
    {!!description && <p> {description} </p>}
    {!!courses && <CourseCardSlider items={courses} />}
    {!!news && <NewsSlider items={news} />}
    {!!button && <ButtonLink>Inscreva-se</ButtonLink>}
  </S.Wrapper>
);

export default ShowCase;
