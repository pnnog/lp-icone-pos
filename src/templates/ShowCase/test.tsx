import '../../../.jest/match-media-mock';
import { screen } from '@testing-library/react';

import ShowCase from '.';
import { renderWithTheme } from 'utils/tests/helper';

describe('<ShowCase />', () => {
  it('should render ShowCase', () => {
    renderWithTheme(<ShowCase />);
  });
});
