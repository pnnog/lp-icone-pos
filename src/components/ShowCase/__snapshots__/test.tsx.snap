// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowCase /> should render ShowCase 1`] = `
.c0 {
  padding-left: calc(9.6rem / 4);
  padding-right: calc(9.6rem / 4);
  width: 100%;
  max-width: 130rem;
  margin-right: auto;
  margin-left: auto;
}

.c1 {
  margin-bottom: 4.8rem;
}

.c1 > p {
  color: #334155;
  font-size: 1.6rem;
  font-weight: 400;
  line-height: 120%;
}

@media (min-width:768px) {
  .c1 {
    margin-bottom: calc(4.8rem * 2 );
  }

  .c1 >p {
    font-size: 2.0rem;
    line-height: 120%;
  }
}

@media (min-width:768px) {

}

@media (max-width:768px) {

}

<div>
  <section
    class="c0 c1"
  >
    <div
      data-testid="Mock Heading"
    />
    <p>
      children test
    </p>
    <div
      data-testid="Mock News Slider"
    />
    <div
      data-testid="Mock Pacient Highlight"
    />
  </section>
</div>
`;
